alloc-customization-point.html

```diff
@@ -165,11 +201,100 @@ <h2>About the customization points, request for feedback/decision</h2>
+Otherwise, <tt>domain_allocator_traits&lt;class T, class Alloc&gt;::value</tt> will be <tt>false</tt>. </p>
+
+<p>Implementation shall provide a definition of <tt>domain_allocator_traits&lt;class T, class Alloc&gt</tt> that is 
+derived from <tt>true_­type</tt> if there exists a function object member <tt>T::domain_alloc_rebind</tt> for 
+which <tt>is_invocable&lt;decltype(T::domain_alloc_rebind), const _Alloc&&gt;::value</tt> is <tt>true</tt>. </p> 
+
+
+
+<p><br/>Modify 20.10.8.1 uses_­allocator trait [allocator.uses.trait]</p>
+
+<p><tt>template &lt;class T, class Alloc&gt; struct uses_allocator;</tt>
+<br/>
+<i>#Remarks:</i> Automatically detects whether <tt>T</tt> has a nested allocator_­type that is convertible from <tt>Alloc</tt>. 
+Meets the <i>Cpp17BinaryTypeTrait</i> requirements ([meta.rqmts]). The implementation shall provide a definition that is 
+derived from <tt>true_­type</tt> if<ins>
+<br/> - </ins>the <i>qualified-id</i> <tt>T::allocator_­type</tt> is valid and denotes a type ([temp.deduct]) and 
+<tt>is_­convertible_­v&lt;Alloc, T::allocator_­type&gt; != false</tt>,<ins> or
+<br/> - if <tt>__domain_allocator_traits&lt;T, Alloc&gt;::value</tt> is <tt>true</tt></ins> 
+<br/><br/>
+<del>otherwise it</del><ins>It</ins> shall be derived from <tt>false_­type</tt><ins> otherwise</ins>. 
+A program may specialize this template to derive from <tt>true_­type</tt> for a program-defined type <tt>T</tt> that does not
+have a nested <tt>allocator_­type</tt> but nonetheless can be constructed with an allocator where either:
+<br/>- the first argument of a constructor has type allocator_­arg_­t and the second argument has type <tt>Alloc</tt> or
+<br/>- the last argument of a constructor has type <tt>Alloc</tt>.
+
+</p>
+
+
+<p>
+Modify [allocator.uses.construction]/p1
+<blockquote>
+ <p>
+<i>Uses-allocator construction</i> with allocator <tt>Alloc</tt> and constructor arguments <tt>args...</tt> refers to the construction
+of an object of type <tt>T</tt> such that <tt>Alloc</tt> is passed to the constructor of T if T uses an allocator type compatible
+with <tt>Alloc</tt><ins>  or <tt>__domain_allocator_traits&lt;T, Alloc&gt;::value</tt> is <tt>true</tt></ins>.
+</p></blockquote>
+</p>
+
+<p>
+Modify [allocator.uses.construction]/p2
+<blockquote>
+The following utility functions support <del>three</del><ins>four</ins> conventions for passing <tt>alloc</tt> to a constructor:
+ ...<br/>
+ <p>
+<ins>- Otherwise, if <tt>__domain_allocator_traits&lt;T, Alloc&gt;::value</tt> is <tt>true</tt>,
+uses-allocator construction invokes T(args..., __domain_allocator_traits&lt;T, Alloc&gt;::rebind(alloc))</ins>
+</p></blockquote>
+</p>
+
+<p>
+Modify [allocator.uses.construction]/p5
+<blockquote>
+ <p>
+ <i>Returns:</i> A tuple value determined as follows:<br/>
+ ...<br/>
+<ins> - Otherwise, if <tt>__domain_allocator_traits&lt;T, Alloc&gt;::value</tt> is <tt>true</tt>, return <tt>forward_as_tuple(std::forward&lt;Args&gt;(args)..., 
```

Comment:
As above, domain-allocator construction needs to be tried first. The program needs to be ill-formed if a nested domain_allocator_type exists but the construction isn't well-formed, and the program needs to be ill-formed if
domain_allocator_traits<T, Alloc>::value is true but the construction isn't well-formed. If the nested domain_allocator_type doesn't exist and domain_allocator_traits<T, Alloc>::value is false, we go to the next bullet,
otherwise return as specified here.

Reply:
ack, added stuff around this change to make it obvious it's the second bullet